crates/web-plugins/didcomm-messaging/protocol-plugins/protocole-handler/src/handler.rs

```diff
@@ -0,0 +1,289 @@
+use std::collections::{HashMap, HashSet};
+use std::sync::{Arc, Mutex};
+
+use plugin_api::{Plugin, PluginError};
+
+use crate::plugins::PLUGINS;
+
+#[derive(Debug, PartialEq)]
+pub enum PluginContainerError {
+    DuplicateEntry,
+    Unloaded,
+    PluginErrorMap(HashMap<String, PluginError>),
+}
+
+pub struct PluginContainer<'a> {
+    loaded: bool,
+    message_type_handlers: HashMap<String, Box<dyn Fn(&str) -> String + Send + Sync>>,
+    plugins: &'a Vec<Arc<Mutex<dyn Plugin>>>,
```

Comment:
You consider message type handlers to have this signature:

```rust 
Fn(&str) -> String + Send + Sync
```

- How sure are you of this? Did you check the signature of current handlers?
- Also note that it is better to organize this concept around a plugin trait.
- How is this plugins property meaningful to you as is? `plugins: &'a Vec<Arc<Mutex<dyn Plugin>>>,`

Reply:
- There is already a plugin trait, and when i am trying to add it is causing some other errors
-  The property plugins: &'a Vec<Arc<Mutex<dyn Plugin>>> indicates a reference to a vector of plugin instances wrapped in an Arc<Mutex<...>, allowing for shared ownership and thread-safe access. it  makes it possible to maintain and interact with multiple plugins concurrently.